sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/storage/block/FileCompactBlockRepository.kt

```diff
@@ -18,51 +18,61 @@ import cash.z.ecc.android.sdk.jni.findBlockMetadata
 import cash.z.ecc.android.sdk.jni.getLatestBlockHeight
 import cash.z.ecc.android.sdk.jni.rewindBlockMetadataToHeight
 import cash.z.ecc.android.sdk.model.BlockHeight
-import cash.z.wallet.sdk.internal.rpc.CompactFormats.CompactBlock
+import co.electriccoin.lightwallet.client.model.CompactBlockUnsafe
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.onEach
 import java.io.File
 
 internal class FileCompactBlockRepository(
     private val blocksDirectory: File,
     private val rustBackend: Backend
 ) : CompactBlockRepository {
 
+    private var currentBufferSize = 0
+    private val metaDataBuffer: MutableList<JniBlockMeta> by lazy { mutableListOf() }
+
     override suspend fun getLatestHeight() = rustBackend.getLatestBlockHeight()
 
     override suspend fun findCompactBlock(height: BlockHeight) = rustBackend.findBlockMetadata(height)
 
-    override suspend fun write(result: Sequence<CompactBlock>): Int {
-        var count = 0
-
-        val metaDataBuffer = mutableListOf<JniBlockMeta>()
-
-        result.forEach { block ->
-            val tmpFile = block.createTemporaryFile(blocksDirectory)
-            // write compact block bytes
-            tmpFile.writeBytesSuspend(block.toByteArray())
-            // buffer metadata
-            metaDataBuffer.add(block.toJniMetaData())
-
-            val isFinalizeSuccessful = tmpFile.finalizeFile()
-            check(isFinalizeSuccessful) {
-                "Failed to finalize file: ${tmpFile.absolutePath}"
+    override suspend fun write(blocks: Flow<CompactBlockUnsafe>): Flow<Int> {
```

Comment:
If we're returning a flow, I don't think the `suspend` is necessary anymore.  At the same time, I do have some concern about returning a flow in this case.  The implementation is correct, but it has potential to be a sharp corner in the future.

Since the flow has side effects in the database and filesystem (it doesn't just emit a final value), collecting the flow multiple times would cause those side effects to occur multiple times.  We don't often invoke a flow for their side effects, but rather for their final value which is where it could be a surprise for a future person working on this code.

Would it make more sense to keep the suspend modifier for the function, do the collection internally (e.g. make the write() method perform side effects) rather than defer the side effects to when the Flow<Int> is collected?